cf-agent/verify_new_packages.c

```diff
@@ -0,0 +1,160 @@
+/*
+   Copyright (C) CFEngine AS
+
+   This file is part of CFEngine 3 - written and maintained by CFEngine AS.
+
+   This program is free software; you can redistribute it and/or modify it
+   under the terms of the GNU General Public License as published by the
+   Free Software Foundation; version 3.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program; if not, write to the Free Software
+  Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA
+
+  To the extent this program is licensed as part of the Enterprise
+  versions of CFEngine, the applicable Commercial Open Source License
+  (COSL) may apply to this file if you as a licensee so wish it. See
+  included file COSL.txt.
+*/
+
+#include <verify_new_packages.h>
+#include <package_module.h>
+#include <logging.h>
+#include <string_lib.h>
+#include <locks.h>
+#include <ornaments.h>
+
+static bool NewPackagePromiseSanityCheck(Attributes a)
+{
+    if (!a.new_packages.module_body || !a.new_packages.module_body->name)
+    {
+        Log(LOG_LEVEL_ERR, "Can not find package module body in policy.");
+        return false;
+    }
+    
+    if (a.new_packages.module_body->updates_ifelapsed == CF_NOINT ||
+        a.new_packages.module_body->installed_ifelapsed == CF_NOINT)
+    {
+        Log(LOG_LEVEL_ERR, "Package module body constraints error.");
+        return false;
+    }
+    
+    if (a.new_packages.package_policy == NEW_PACKAGE_ACTION_NONE)
+    {
+        Log(LOG_LEVEL_ERR, "Not supported package policy in new package"
+            " promise.");
```

Comment:
``s/Not supported/Unsupported/``

``s/new package promise/package promise/`` - The fact that it's new quickly becomes obsolete.